### PR TITLE
feat: Make TableFrame in Kai use schema type to generate Columns

### PR DESCRIPTION
**Pull request by mykola-vrmchk**

### CODE REVIEW

**Comment by mykola-vrmchk** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2425** <a href="https://app.graphite.dev/github/pr/dxos/dxos/2425" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dxos/dxos/2425?utm_source=stack-comment).

File: packages/experimental/kai/src/frames/TableFrame.tsx [hunk 1]
```diff
@@ -5,68 +5,57 @@
...
-  }
-];
+const capitalizeFirstLetter = (str: string) => {
+  return str.charAt(0).toUpperCase() + str.slice(1);
+};
 
-const getType = (id: string): ColumnType<any> => types.find((type) => type.id === id)!;
+const generateTypes = (schemas: EchoSchemaType[]) => {
+  const generateColumns = (schema: EchoSchemaType) => {
+    const basicTypes = ['string', 'number', 'boolean'];
```
Comment (dmaretskyi) [inline packages/experimental/kai/src/frames/TableFrame.tsx:27]: move to module level.

rename SUPPORTED_TYPES

File: packages/experimental/kai/src/frames/TableFrame.tsx [hunk 2]
```diff
@@ -5,68 +5,57 @@
...
-      { Header: 'ZIP', accessor: 'address.zip' as any }
-    ]
-  }
-];
+const capitalizeFirstLetter = (str: string) => {
+  return str.charAt(0).toUpperCase() + str.slice(1);
+};
 
-const getType = (id: string): ColumnType<any> => types.find((type) => type.id === id)!;
+const generateTypes = (schemas: EchoSchemaType[]) => {
```
Comment (dmaretskyi) [inline packages/experimental/kai/src/frames/TableFrame.tsx:25]:
```suggestion
const generateTypes = (schemaTypes: EchoSchemaType[]) => {
```

File: packages/experimental/kai/src/frames/TableFrame.tsx [hunk 3]
```diff
@@ -5,68 +5,57 @@
...
+  const generateColumns = (schema: EchoSchemaType) => {
```
Comment (dmaretskyi) [inline packages/experimental/kai/src/frames/TableFrame.tsx:26]:
```suggestion
  const generateColumns = (type: EchoSchemaType) => {
```

**Pull request merged**